modules/xnft/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+// This file is part of Acala.
+
+// Copyright (C) 2023 Unique Network.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+use frame_support::{ensure, pallet_prelude::*};
+use frame_system::pallet_prelude::*;
+use module_nft::{ClassIdOf, TokenIdOf};
+use sp_runtime::{traits::AccountIdConversion, DispatchResult};
+use sp_std::boxed::Box;
+use xcm::v3::{
+	AssetId, AssetInstance, Error as XcmError, Fungibility, Junction::*, MultiAsset, MultiLocation, Result as XcmResult,
+};
+use xcm_executor::traits::{ConvertLocation, Error as XcmExecutorError, TransactAsset};
+
+pub mod impl_transactor;
+pub mod xcm_helpers;
+
+pub use pallet::*;
+
+pub type ConverterOf<T> = <T as Config>::LocationToAccountId;
+pub type ModuleNftPallet<T> = module_nft::Pallet<T>;
+pub type OrmlNftPallet<T> = orml_nft::Pallet<T>;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use primitives::nft::{ClassProperty, Properties};
+
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + module_nft::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		type LocationToAccountId: ConvertLocation<Self::AccountId>;
+
+		type NtfPalletLocation: Get<MultiLocation>;
+
+		type RegisterOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+	}
+
+	/// Error for non-fungible-token module.
+	#[pallet::error]
+	pub enum Error<T> {
+		AssetAlreadyRegistered,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		AssetRegistered {
+			asset_id: AssetId,
+			collection_id: ClassIdOf<T>,
+		},
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn assets)]
```

Comment:
Я бы предложил убрать proxy-функции, которые только содержат вызов функции с другим именем. И назвать просто нужную функцию как надо напрямую.

```suggestion
	#[pallet::getter(fn foreign_asset_to_collection)]
```

Reply:
И хотелось бы переименовать стореджи похожим образом. На глаз в коде эти `classes` и `assets` плохо читаются. Хочется как-то вставить слова "foreign asset(s)" туда